src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/XMLPPIO.java

```diff
@@ -76,6 +80,33 @@ public void encode( Object value, OutputStream os) throws Exception {
     public String getFileExtension() {
         return "xml";
     }
+    
+    /**
+     * Returns a parser using the given configuration
+     * 
+     * @param configuration
+     * @return
+     */
+    protected Parser getParser(Configuration configuration) {
+        Parser parser = new Parser(configuration);
+        // spring injection would have been preferred, but it changes the order in which PPIO
+        // are listed, which in turn chances the behavior of WPS (e.g., default mimeType)
+        parser.setEntityResolver(getResolveProvider().getEntityResolver());
+        return parser;
+    }
```

Comment:
So this makes sense, the fundamental change is supplying a EntryResolver to parsers before use.